proposed-solutions.md

```diff
@@ -39,8 +69,48 @@ You write external metadata in a `package.json` to determine file type.
 }
 ```
 
-#### 6. extension (".mjs")
+More can be read about this proposal
+[here](https://github.com/dherman/defense-of-dot-js/blob/master/proposal.md).
+
+* While this removes all ambiguity at parse time, it does so by tightly
+  coupling files to a corresponding `package.json`. In order to determine
+  the module type of a given file, one must examine every directory up from
+  the file's location. For symlinked directories, what constitutes "up" may
+  depend on how the file is being viewed.
+* Moving a JS file from one directory to another may change how it is parsed
+  without any other action being taken on the file, by removing it from a path
+  including `package.json`.
+* `"modules.root"` introduces an abstraction between `require('x/y')` that may
+  result in `require('x/src/y')` being required, which may be difficult for
+  beginners to follow.
+* Care must be taken in implementing `"modules.root"` to avoid allowing the
+  new package root to be set outside of the current package.
+* It does not solve for standalone JS files.
+
+#### 6. File Extension (".mjs")
+
+Use a new extension to indicate whether a file is an ES module (`.mjs`) or a
+Node.js module (`.js`).
+
+* This requires the registration of a new MIME type.
```

Comment:
Why? The browser loader spec determines the file type from how the module was requested, so browsers don't use MIME information. I know that @bradleymeck did this, but I don't know why it was required, so I doubt most other people do either.